elide-datastore/elide-datastore-aggregation/src/test/java/com/yahoo/elide/datastores/aggregation/dimension/DimensionTest.java

```diff
@@ -0,0 +1,102 @@
+            "name"
+    );
+
+    private static final Dimension DEGENERATE_DIMENSION = new DegenerateDimension(
+            "overallRating",
+            null,
+            String.class,
+            CardinalitySize.SMALL,
+            "overallRating",
+            ColumnType.FIELD
+    );
+
+    private static final Dimension TIME_DIMENSION = new TimeDimension(
+            "recordedTime",
+            null,
+            Long.class,
+            CardinalitySize.LARGE,
+            "recordedTime",
+            TimeZone.getDefault(),
+            TimeGrain.DAY
+    );
+
+    @Test
+    public void testDimensionAsCollectionElement() {
+        Assert.assertEquals(ENTITY_DIMENSION, ENTITY_DIMENSION);
+        Assert.assertEquals(DEGENERATE_DIMENSION, DEGENERATE_DIMENSION);
+        Assert.assertNotEquals(ENTITY_DIMENSION, DEGENERATE_DIMENSION);
+        Assert.assertNotEquals(ENTITY_DIMENSION.hashCode(), DEGENERATE_DIMENSION.hashCode());
+
+        // different dimensions should be separate elements in Set
+        Set<Dimension> dimensions = new HashSet<>();
+        dimensions.add(ENTITY_DIMENSION);
+
+        Assert.assertEquals(dimensions.size(), 1);
+
+        // a separate same object doesn't increase collection size
+        Dimension sameEntityDimension = new EntityDimension(
+                "country",
+                null,
+                Country.class,
+                CardinalitySize.SMALL,
+                "name"
+        );
+        Assert.assertEquals(sameEntityDimension, ENTITY_DIMENSION);
+        dimensions.add(sameEntityDimension);
+        Assert.assertEquals(dimensions.size(), 1);
+
+        dimensions.add(ENTITY_DIMENSION);
+        Assert.assertEquals(dimensions.size(), 1);
+
+        dimensions.add(DEGENERATE_DIMENSION);
+        Assert.assertEquals(dimensions.size(), 2);
+
+        dimensions.add(TIME_DIMENSION);
+        Assert.assertEquals(dimensions.size(), 3);
+    }
+
+    @Test
+    public void testToString() {
+        // table dimension
+        Assert.assertEquals(
+                ENTITY_DIMENSION.toString(),
+                "EntityDimension[name='country', longName='country', description='country', dimensionType=TABLE, dataType=Country, cardinality=SMALL, friendlyName='name']"
+        );
+
+        // degenerate dimension
+        Assert.assertEquals(
+                DEGENERATE_DIMENSION.toString(),
+                "DegenerateDimension[columnType=FIELD, name='overallRating', longName='overallRating', description='overallRating', dimensionType=DEGENERATE, dataType=String, cardinality=SMALL, friendlyName='overallRating']"
+        );
+
+        Assert.assertEquals(
+                TIME_DIMENSION.toString(),
+                "TimeDimension[timeZone=Pacific Standard Time, timeGrain=DAY, columnType=TEMPORAL, name='recordedTime', longName='recordedTime', description='recordedTime', dimensionType=DEGENERATE, dataType=class java.lang.Long, cardinality=LARGE, friendlyName='recordedTime']"
```

Comment:
The timezone "Pacific Standard Time" cause the test to fail in my local testing as I am in CST.

Reply:
Thanks. It's a bug in test setup. Try again